romancal/pipeline/tests/test_exposure_pipeline.py

```diff
@@ -53,3 +54,45 @@ def test_input_to_output(input_value, expected_output_type):
     [setattr(getattr(pipeline, k), "skip", True) for k in pipeline.step_defs]
     output_value = pipeline.run(input_value)
     assert isinstance(output_value, expected_output_type)
+
+
+@pytest.mark.parametrize("save_results", [True, False])
+@pytest.mark.parametrize("skip_tweakreg", [True, False])
+def test_elp_save_results(
+    function_jail, tmp_path, save_results, skip_tweakreg, monkeypatch
+):
+    """
+    Test that the elp respects save_results.
+    """
+    output_path = tmp_path / "output"
+    output_path.mkdir()
+
+    model = rdm.ScienceRawModel.create_fake_data()
+    model.meta.filename = "test_uncal.asdf"
+    model.meta.exposure.read_pattern = [[1], [2]]  # truncated for 2 groups below
+    model.meta.exposure.ma_table_number = 5
+    model.data = np.zeros((2, 4096, 4096), dtype=model.data.dtype)
+    model.amp33 = np.zeros((2, 4096, 128), dtype=model.amp33.dtype)
+
+    pipeline = ExposurePipeline()
+    pipeline.output_dir = str(output_path)
+    pipeline.save_results = save_results
+
+    if skip_tweakreg:
+        pipeline.tweakreg.skip = True
+    else:
+        # don't try to actually run tweakreg as it will fail for an empty model
+        monkeypatch.setattr(pipeline.tweakreg, "run", lambda init: init)
+
+    pipeline.run(model)
+    # check that the current directory doesn't contain extra files
+    assert set(p.name for p in tmp_path.iterdir()) == {"output"}
+
+    output_files = set(p.name for p in output_path.iterdir())
+    # TODO shouldn't this be empty?
+    expected = {"test_segm.asdf", "test_cat.parquet"}
```

Comment:
@schlafly Would you consider it a bug that the pipeline produces these files even when `save_results==False`?

Reply:
Thanks! I opened up a follow-up issue with the details you provided. Discussion of possible alternatives can be discussed there.

https://github.com/spacetelescope/romancal/issues/1960